File: Assets/Prefabs/Zombie/ZombieAttackingAIController.js
#pragma strict

var damage = 1F;
var sounds : AudioClip[];

private var playerHealthController : EntityHealthController;
private var audioSource : AudioSource;

function Start() {
	playerHealthController = GameController.instance.player.GetComponent(EntityHealthController);
	audioSource = GetComponent(AudioSource);
}

function Attack() {
	playerHealthController.Damage(damage);
	RandomSoundPlayer();
}

function RandomSoundPlayer() {
	audioSource.clip = sounds[Random.Range(0,sounds.length)];
	audioSource.Play();
}

function OnChangeAIStrategy( strategy : String ) {
	enabled = (strategy == "Attacking");
	
	if(enabled) {
		gameObject.animation.Play("Eating");
		InvokeRepeating("Attack", 1, 1);
	} else {
		CancelInvoke();
	}
}

File: Assets/Prefabs/Zombie/ZombieAttackingAIController.cs
using UnityEngine;
using System.Collections;

public class ZombieAttackingAIController : MonoBehaviour {
	public int damage = 1;
	public AudioClip[] sounds;
	
	private EntityHealthController playerHealthController;
	private AudioSource audioSource;
	
	void Awake() {
		playerHealthController = GameController.instance.player.GetComponent<EntityHealthController>();
		audioSource = GetComponent<AudioSource>();
	}
	
	public void Attack() {
		playerHealthController.Damage(damage);
		RandomSoundPlayer();
	}
	
	private void RandomSoundPlayer() {
		audioSource.clip = sounds[Random.Range(0,sounds.Length - 1)];
		audioSource.Play();
	}
	
	public void OnChangeAIStrategy( string strategy ) {
		enabled = (strategy == "Attacking");
		
		if(enabled) {
			gameObject.animation.Play("Eating");
			InvokeRepeating("Attack", 0, 1);
		} else {
			CancelInvoke();
		}
	}
}
